app/src/main/java/com/knocklock/di/DispatcherModule.kt

```diff
@@ -0,0 +1,47 @@
+package com.knocklock.di
+
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import javax.inject.Qualifier
+
+
+@Module
+@InstallIn(SingletonComponent::class)
+class DispatcherModule {
+
+    @Provides
+    @IODispatcher
+    fun provideDispatcherIO(): CoroutineDispatcher = Dispatchers.IO
+
+    @Provides
+    @MainDispatcher
+    fun provideDispatcherMain(): CoroutineDispatcher = Dispatchers.Main
+
+    @Provides
+    @DefaultDispatcher
+    fun provideDispatcherDefault(): CoroutineDispatcher = Dispatchers.Default
+
+    @Provides
+    @MainImmediateDispatcher
+    fun provideDispatcherMainImmediate(): CoroutineDispatcher = Dispatchers.Main.immediate
```

Comment:
```suggestion
    @Provides
    @Singleton
    @IODispatcher
    fun provideDispatcherIO(): CoroutineDispatcher = Dispatchers.IO

    @Provides
    @Singleton
    @MainDispatcher
    fun provideDispatcherMain(): CoroutineDispatcher = Dispatchers.Main

    @Provides
    @Singleton
    @DefaultDispatcher
    fun provideDispatcherDefault(): CoroutineDispatcher = Dispatchers.Default

    @Provides
    @Singleton
    @MainImmediateDispatcher
    fun provideDispatcherMainImmediate(): CoroutineDispatcher = Dispatchers.Main.immediate
```

Singleton 어노테이션 달아줘야 할 것 같아요!